Unpinned dependency for gulp-clean-css breaks our CSS

It looks like a new version of [`gulp-clean-css`](https://www.npmjs.com/package/gulp-clean-css), 2.4.0, was released in the past day or two.  The previous version was 2.3.2, but our `package.json` specifies its version requirement as `^2.0.11`, so any brand-new `npm install`s will use 2.4.0.

Upon installation, it displays the following:

```
npm WARN deprecated gulp-clean-css@2.4.0: breaking changes from clean-css 4.x. Please install gulp-clean-css 3.x
```

Looking at the package's [`CHANGELOG.md`](https://github.com/scniro/gulp-clean-css/blob/master/CHANGELOG.md), it appears the maintainer *accidentally* released 2.4.0 as a minor version upgrade, when in fact they meant to release 3.0.0 because of the breaking change.

The effect of this breaking change is that `frontend.tests.test_collectstatic.CompressedCollectStaticTests` is now broken; specifically, relative URLs are mangled such that we can't ascend the directory tree beyond one level with `..`. The broken test output looks like this:

```
E                   whitenoise.storage.MissingFileError: The file 'frontend/built/images/info-badge-filled.svg' could not be found with <whitenoise.storage.CompressedManifestStaticFilesStorage object at 0x7fcfa84218d0>.
E
E                   The CSS file 'frontend/built/style/data_explorer.min.css' references a file which could not be found:
E                     frontend/built/images/info-badge-filled.svg
E
E                   Please check the URL references in this CSS file, particularly any
E                   relative paths which might be pointing to the wrong location
```

I believe this breakage hasn't yet shown up on Travis because we've enabled its [dependency caching](https://blog.travis-ci.com/2013-12-05-speed-up-your-builds-cache-your-dependencies).  If we're going to use dependency caching on travis, I think it might be a good idea for us to pin our dependencies more strongly, lest we run into unexpected problems like this only when deploying to production. (I only ran into this because I was working on #1342, which involved installing CALC from scratch.)

--- a/package.json
+++ b/package.json
@@ -26,7 +26,7 @@
     "eslint-plugin-react": "^6.9.0",
     "eyeglass": "1.1.2",
     "gulp": "^3.9.1",
-    "gulp-clean-css": "^2.0.11",
+    "gulp-clean-css": "2.3.2",
     "gulp-concat": "^2.6.0",
     "gulp-eslint": "^3.0.1",
     "gulp-if": "^2.0.1",

